Src/coffeescript/cql-execution/src/elm/aggregate.coffee

```diff
@@ -1,24 +1,32 @@
 { Expression } = require './expression'
 { typeIsArray , allTrue, anyTrue, compact, numerical_sort} = require '../util/util'
 { build } = require './builder'
+{ Exception } = require '../datatypes/exception'
 Quantity = require './quantity'
 
 quantitiesOrArg = (arr) ->
-  if arr[0]?.constructor.name == "Quantity"
+  arr = compact(arr)
+  # short curcuit empty arrays and return
+  if arr.length == 0
+    return arr
+
+  allQs = arr.every (x) -> x.constructor.name == "Quantity"
+  someQs = arr.some (x) -> x.constructor.name == "Quantity"
+  if allQs
     unit = arr[0].unit
     values = []
     for i in arr
-      if i.constructor.name == "Quantity" && i.unit == unit
-        values.push i.value
-      else
-        return []
+      values.push i.convertUnits(unit)
     return compact(values) # need to make sure that there are not any null values from the quntities
```

Comment:
Yeah, now that I've looked through the code more thoroughly, I don't think `i.convertUnits` can return `null` -- so if you're editing the code anyway, I say remove the extra `compact`.  Especially since this function gets called a lot.